src/propagation/serialization.jl

```diff
@@ -0,0 +1,204 @@
+@doc raw"""
+    TaylorInterpolantNSerialization{T}
+
+Custom serialization struct to save a `TaylorInterpolant{T, TaylorN{T}, 2}` to a `.jld2` file. 
+
+# Fields
+- `vars::Vector{String}`: jet transport variables. 
+- `order::Int`: order of Taylor polynomials.
+- `varorder::Int`: order of jet transport perturbations. 
+- `dims::Tuple{Int, Int}`: matrix dimensions. 
+- `t0::T`: initial time.
+- `t::Vector{T}`: vector of times. 
+- `x::Vector{T}`: vector of coefficients. 
+"""
+struct TaylorInterpolantNSerialization{T}
+    vars::Vector{String}
+    order::Int
+    varorder::Int
+    dims::Tuple{Int, Int}
+    t0::T
+    t::Vector{T}
+    x::Vector{T}
+end 
+
+# Tell JLD2 to save TaylorInterpolant{T, TaylorN{T}, 2} as TaylorInterpolantNSerialization{T}
+writeas(::Type{TaylorInterpolant{T, TaylorN{T}, 2}}) where {T} = TaylorInterpolantNSerialization{T}
+
+# Convert method to write .jld2 files
+function convert(::Type{TaylorInterpolantNSerialization{T}}, eph::TaylorInterpolant{T, TaylorN{T}, 2}) where {T}
+    # Variables 
+    vars = TS.get_variable_names()
+    # Number of variables 
+    n = length(vars)
+    # Matrix dimensions
+    dims = size(eph.x)
+    # Number of elements in matrix
+    N = dims[1] * dims[2]
+    # Taylor1 order
+    order = eph.x[1, 1].order
+    # Number of coefficients in each Taylor1
+    k = order + 1
+    # TaylorN order 
+    varorder = eph.x[1, 1].coeffs[1].order
+    # Number of coefficients in each TaylorN
+    L = varorder + 1
+    # Number of coefficients in each HomogeneousPolynomial
+    M = binomial(n + varorder, varorder)
+    # M = sum(binomial(n + i_3 - 1, i_3) for i_3 in 0:varorder)
+    
+    # Vector of coefficients 
+    x = Vector{T}(undef, N * k * M)
+
+    # Save coefficients 
+    i = 1
+    # Iterate over matrix elements 
+    for i_1 in 1:N
+        # Iterate over Taylor1 coefficients
+        for i_2 in 1:k
+            # Iterate over TaylorN coefficients
+            for i_3 in 0:varorder
+                # Iterate over i_3 order HomogeneousPolynomial
+                for i_4 in 1:binomial(n + i_3 - 1, i_3)
+                    x[i] = eph.x[i_1].coeffs[i_2].coeffs[i_3+1].coeffs[i_4]
+                    i += 1
+                end 
+            end 
+        end 
+    end 
+
+    return TaylorInterpolantNSerialization{T}(vars, order, varorder, dims, eph.t0, eph.t, x)
+end 
+
+# Convert method to read .jld2 files
+function convert(::Type{TaylorInterpolant{T, TaylorN{T}, 2}}, eph::TaylorInterpolantNSerialization{T}) where {T} 
+    # Variables 
+    vars = eph.vars
+    # Number of variables 
+    n = length(vars)
+    # Matrix dimensions
+    dims = eph.dims
+    # Number of elements in matrix
+    N = dims[1] * dims[2]
```

Comment:
Same as above.

Reply:
In this case, since we are reconstructing the matrix, there is no `x` to do `length(x)`.